ecosystem/typescript/sdk/src/providers/indexer.ts

```diff
@@ -0,0 +1,82 @@
+import { AnyNumber } from "../bcs/types";
+import axios from "axios";
+import { print } from "graphql";
```

Comment:
`graphql` dependency is needed for code generator. `print` method is used to parse the generated query from `AST` to a `string`.

I was trying to not use it, but that means that instead of writing our queries in `.graphql` files and then use codegen to generate `ts` representation, we write the queries directly in `.ts` files - but then we might lose some `.graphql` syntax benefits like inject fragments 

with graphql file
```
current_token_data {
      ...TokenDataFields
    }
```

with ts file
```
current_token_data {
      creator_address
      collection_name
      description
      metadata_uri
      name
      token_data_id_hash
      collection_data_id_hash
    }
```

Reply:
can you pre-generate those strings if you do need to parse the AST? then you can put graphql as a devDependency.